exercises/practice/prime-factors/.meta/Example.roc

```diff
@@ -0,0 +1,18 @@
+module [primeFactors]
+
+primeFactors : U64 -> List U64
+primeFactors = \value ->
+    findPrimeFactors = \n, p ->
+        if n < 2 then
+            []
+        else if n % p == 0 then
+            findPrimeFactors (n // p) p |> List.prepend p
+        else if p * p < n then
+            nextP = if p == 2 then 3 else p + 2
+            findPrimeFactors n nextP
+        else
+            [n]
+
+    when findPrimeFactors value 2 is
+        [] -> if value < 2 then [] else [value]
+        result -> result
```

Comment:
```suggestion
    findPrimeFactors = \factors, n, p ->
        if n < 2 then
            factors
        else if n |> Num.isMultipleOf p then
            findPrimeFactors (List.append factors p) (n // p) p
        else if p * p < n then
            nextP = if p == 2 then 3 else p + 2
            findPrimeFactors factors n nextP
        else
            List.append factors n

    findPrimeFactors [] value 2
```
What would you think about doing something like this? The most significant change here is to explicitly pass in a list at the beginning of the recursive function and append to that rather than prepending to build up the result. This way the original input list can be mutated behind the scenes each time instead of having to copy the whole list each time a new factor is prepended. Obviously the performance doesn't actually matter in this scenario, but I think it is worth highlighting this style given that Roc doesn't use linked lists and does in place mutation.